plugins/woocommerce/includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -1144,7 +1163,16 @@ public function find_matching_product_variation( $product, $match_attributes = a
 
 			// Loop over the variation meta keys and values i.e. what is saved to the products. Note: $attribute_value is empty when 'any' is in use.
 			foreach ( $variation as $attribute_key => $attribute_value ) {
-				$match_any_value = '' === $attribute_value;
+				/**
+				 * Filter whether it should match any value or not.
+				 *
+				 * @param  bool   $match_any_value The result of '' === $attribute_value.
+				 * @param  string $attribute_value The attribute value.
+				 * @param  string $attribute_key   The attribute key.
+				 * @param  int    $variation_id    The variation ID.
+				 * @return bool
+				 */
+				$match_any_value = apply_filters( 'woocommerce_match_any_value', '' === $attribute_value, $attribute_value, $attribute_key, $variation_id );
```

Comment:
Hi, peterfabian.

I like `woocommerce_find_matching_product_variation_match_any` since it follows the same pattern than the SQL query filter (`woocommerce_find_matching_product_variation_query`)